src/flash.rs

```diff
@@ -116,18 +115,3 @@ pub fn wait_for_idle() -> i32 {
 
     0
 }
-
-#[cfg(feature = "esp32s2")]
-unsafe fn esp_rom_spiflash_erase_chip() -> i32 {
```

Comment:
I don't know where I got this from, probably misunderstood the ESP32-specific patch reimplementation of the rom function. Either way, this was missing the bit to disable write protection...